src/lib.rs

```diff
@@ -113,13 +114,16 @@ pub fn process(ctx: &Ctx) -> Result<()> {
     let source_files = template_composite.find_sourcefiles()?;
     debug!("defining plan of rendering");
     let actions = plan(ctx, source_files, &used_variables)?;
+
     if ui::confirm_plan(ctx, &actions)? {
         debug!("executing plan of rendering");
-        execute(ctx, &actions, &used_variables)?;
-        debug!("Saving metadata");
-        timeline::save_options(&used_variables, &ctx.cmd_opt.src, &ctx.cmd_opt.dst_folder)?;
+        let new_metas = execute(ctx, &actions, &used_variables)?;
         debug!("running scripts");
         run_scripts(ctx, &template_composite)?;
+        debug!("Saving metadata");
+        timeline::save_options(&used_variables, &ctx.cmd_opt.src, &ctx.cmd_opt.dst_folder)?;
+
+        timeline::save_metas_for_source(new_metas, &ctx.cmd_opt.dst_folder, "global".into())?;
```

Comment:
🤌 **nitpick**: If you transform a `&str` to a `String`, You can use `String::from("global")` or `"global".to_string()` to be more explicit